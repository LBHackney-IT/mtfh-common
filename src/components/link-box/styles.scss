.lbh-link-overlay a {
  position: static;

  &:before {
    content: '';
    cursor: inherit;
    display: block;
    position: absolute;
    top: 0;
    left: 0;
    z-index: 0;
    width: 100%;
    height: 100%;
  }
}

.lbh-link-box {
  position: relative;

  a[href]:not(.lbh-link-box a),
  abbr[title] {
    position: relative;
    z-index: 1;
  }
}
