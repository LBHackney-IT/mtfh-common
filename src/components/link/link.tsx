import React, { ComponentPropsWithoutRef, forwardRef } from 'react';
import cn from 'classnames';
import { navigateToUrl } from 'single-spa';

import './styles.scss';

export interface LinkProps extends ComponentPropsWithoutRef<'a'> {
  variant?:
    | 'link'
    | 'danger'
    | 'text-colour'
    | 'muted'
    | 'back-link'
    | 'native';
  isExternal?: boolean;
}

export const Link = forwardRef<HTMLAnchorElement, LinkProps>(function Link(
  {
    variant = 'link',
    isExternal = false,
    href,
    className,
    children,
    rel,
    target,
    ...props
  },
  ref
) {
  const linkClasses = cn(
    variant !== 'native' && {
      'govuk-link lbh-link': variant !== 'back-link',
      'govuk-back-link lbh-back-link': variant === 'back-link',
      [`lbh-link--${variant}`]: variant !== 'link' && variant !== 'back-link',
    },
    className
  );
  return (
    // eslint-disable-next-line react/jsx-no-target-blank
    <a
      ref={ref}
      href={href}
      className={linkClasses}
      rel={isExternal ? 'noopener noreferrer' : rel}
      target={isExternal ? '_blank' : target}
      onClick={!isExternal ? navigateToUrl : undefined}
      {...props}
    >
      {children}
    </a>
  );
});
