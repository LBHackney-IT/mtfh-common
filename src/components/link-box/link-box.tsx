import React, {
  ComponentPropsWithoutRef,
  ReactElement,
  forwardRef,
} from 'react';
import cn from 'classnames';

import { ButtonLinkProps } from '../button-link';
import { LinkProps } from '../link';
import './styles.scss';

export interface LinkOverlayProps extends ComponentPropsWithoutRef<'div'> {
  children: ReactElement<LinkProps | ButtonLinkProps>;
}

export const LinkOverlay = forwardRef<HTMLDivElement, LinkOverlayProps>(
  function LinkOverlay({ children, className, ...props }, ref) {
    return (
      <div ref={ref} className={cn('lbh-link-overlay', className)} {...props}>
        {children}
      </div>
    );
  }
);

export const LinkBox = forwardRef<
  HTMLDivElement,
  ComponentPropsWithoutRef<'div'>
>(function LinkBox({ children, className, ...props }, ref) {
  return (
    <div ref={ref} className={cn('lbh-link-box', className)} {...props}>
      {children}
    </div>
  );
});
