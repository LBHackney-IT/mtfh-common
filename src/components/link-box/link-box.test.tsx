import React from 'react';
import { render } from '@testing-library/react';

import { Link } from '../link';
import { LinkBox, LinkOverlay } from './link-box';

test('it renders correctly', () => {
  const { container } = render(
    <LinkBox>
      <LinkOverlay>
        <Link href="https://localhost">Link</Link>
      </LinkOverlay>
    </LinkBox>
  );

  expect(container).toMatchSnapshot();
});
