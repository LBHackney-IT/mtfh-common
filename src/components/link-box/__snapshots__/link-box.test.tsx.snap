// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div>
  <div
    class="lbh-link-box"
  >
    <div
      class="lbh-link-overlay"
    >
      <a
        class="govuk-link lbh-link"
        href="https://localhost"
      >
        Link
      </a>
    </div>
  </div>
</div>
`;
